src/scripts/main.js

```
@@ -1,3 +1,117 @@
 'use strict';
 
 // write code here
+
+const table = document.querySelector('.field');
+const appendRowButton = document.querySelector('.append-row');
+const removeRowButton = document.querySelector('.remove-row');
+const appendColButton = document.querySelector('.append-column');
+const removeColButton = document.querySelector('.remove-column');
```

Add a null-check / guard for the required DOM elements selected at the top. Right now you query `table`, `appendRowButton`, `removeRowButton`, `appendColButton`, and `removeColButton` and then use them directly (addEventListener, updateButtonsState). If any selector returns `null` the script will throw and tests/environments without expected markup will fail with an unclear error. Consider adding an early guard after these queries to fail fast or return early, for example:

```javascript
if (!table || !appendRowButton || !removeRowButton || !appendColButton || !removeColButton) {
  throw new Error('Required DOM elements not found');
}
```

(or return early). This makes initialization safer and errors easier to diagnose.